xbmc/windowing/gbm/DRMAtomic.cpp

```diff
@@ -22,7 +24,52 @@
 #include <unistd.h>
 
 using namespace KODI::WINDOWING::GBM;
-
+const std::string SETTING_VIDEOSCREEN_HW_SCALING_FILTER = "videoscreen.hwscalingfilter";
+enum drm_scaling_filter
+{
+  DRM_SCALING_FILTER_DEFAULT,
+  DRM_SCALING_FILTER_NEAREST_NEIGHBOR,
+};
+uint32_t CDRMAtomic::GetScalingFilterType(const char* type)
+{
+  if (!strcmp(type, "Nearest Neighbor"))
+    return DRM_SCALING_FILTER_NEAREST_NEIGHBOR;
+  else
+    return DRM_SCALING_FILTER_DEFAULT;
+}
+uint32_t CDRMAtomic::GetScalingFactor(uint32_t srcWidth,
+                                      uint32_t srcHeight,
+                                      uint32_t destWidth,
+                                      uint32_t destHeight)
+{
+  uint32_t factor_W{1};
+  uint32_t factor_H{1};
+  factor_W = destWidth / srcWidth;
```

Comment:
you can do this in the initialization, no need to add extra lines

Reply:
yes, I will change accordingly.